gobblin-utility/src/main/java/gobblin/util/AvroUtils.java

```diff
@@ -156,4 +170,116 @@ public static GenericRecord convertRecordSchema(GenericRecord record, Schema new
       closer.close();
     }
   }
+
+  /**
+   * Get the latest avro schema for a directory
+   * @param directory the input dir that contains avro files
+   * @param conf configuration
+   * @param latest true to return latest schema, false to return oldest schema
+   * @return the latest/oldest schema in the directory
+   * @throws IOException
+   */
+  public static Schema getDirectorySchema(Path directory, Configuration conf, boolean latest) throws IOException {
+    Schema schema = null;
+    Closer closer = Closer.create();
+    try {
+      List<FileStatus> files = getDirectorySchemaHelper(directory, FileSystem.get(conf));
+      FileStatus file = latest ? files.get(0) : files.get(files.size() - 1);
+      LOG.info("Path to get the avro schema: " + file);
+      FsInput fi = new FsInput(file.getPath(), conf);
+      GenericDatumReader<GenericRecord> genReader = new GenericDatumReader<GenericRecord>();
+      schema = new DataFileReader<GenericRecord>(fi, genReader).getSchema();
+    } catch (IOException e) {
+      throw new IOException("Cannot get the schema for directory " + directory);
```

Comment:
please add the parent exception so we can preserve the stack trace: throw new IOException("Cannot get the schema for directory " + directory, e);